### PR TITLE
[MOSIP-28486] Added documentation for all the reusable workflows

### PR DESCRIPTION
**Pull request by syedsalman3753**
[MOSIP-25631]

[MOSIP-25631]: https://mosip.atlassian.net/browse/MOSIP-25631?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: docs/image-transfer-readme.md [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+This workflow is designed to automate the process of transferring Docker images to a specified Docker Hub destination organization.
+It uses GitHub Actions to facilitate the image transfer.
+
+## Purpose
+The purpose of this workflow is to perform the following tasks:
+
+1. Update the configuration file config.yml with Docker Hub credentials and the destination organization. 
+2. Push Docker images to the destination organization on Docker Hub. 
+3. Notify the status of the workflow via Slack.
```
Comment (ckm007) [inline docs/image-transfer-readme.md:11]: are we notifying?
In case case yes.. use devops only for notifications to be triggered and only after failures.

Comment (syedsalman3753) [reply docs/image-transfer-readme.md:11]: If it fails, then we will notify devops slack group

File: docs/chart-lint-publish-readme.md [hunk 2]
```diff
@@ -0,0 +1,99 @@
+# chart-lint-publish
+
+## Purpose
+This workflow is designed to perform the following tasks:
+1. Validate Helm charts in pull requests by checking for changes in the Helm chart directory.
+2. Generate a list of charts that are modified in the base branch compared to the pull request branch.
+3. Optionally include or exclude specific charts from validation.
+4. Publish Helm charts to a Helm chart repository on successful validation.
+
+## Inputs
```
Comment (SwatiPeriwal) [inline docs/chart-lint-publish-readme.md:10]: Please leave a sentence space and give an introductory line for the section on Input..

File: docs/chart-lint-publish-readme.md [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+1. **Validate Helm Charts in Pull Requests**: Validates Helm charts in pull requests by checking for chart changes. A list of charts is generated.
+    - This step runs when the event is a pull request.
+
+2. **Get Helm Charts List**: Retrieves a list of Helm charts to validate or publish.
+    - This step runs when the event is not a pull request.
+
+3. **Validate / Publish Helm Charts**: Uses the `mosip/helm-gh-pages` action to validate and publish Helm charts.
+    - This step includes linting and chart publication based on the input settings.
+
+## Example Usage
```
Comment (SwatiPeriwal) [inline docs/chart-lint-publish-readme.md:39]: Maintain consistency... leave a line space below as header..

File: docs/codeql-analysis-readme.md [hunk 4]
```diff
@@ -0,0 +1,33 @@
+# CodeQL Analysis Workflow
+
+## Purpose
+This workflow is designed to perform the following tasks:
+1. This GitHub Actions workflow is used for performing CodeQL analysis on the repository.
```
Comment (SwatiPeriwal) [inline docs/codeql-analysis-readme.md:5]: `CodeQL`

File: docs/image-transfer-readme.md [hunk 5]
```diff
@@ -0,0 +1,58 @@
+# Manual Workflow to Transfer Docker Images
+
+This workflow is designed to automate the process of transferring Docker images to a specified Docker Hub destination organization.
+It uses GitHub Actions to facilitate the image transfer.
+
+## Purpose
+The purpose of this workflow is to perform the following tasks:
+
+1. Update the configuration file config.yml with Docker Hub credentials and the destination organization.
```
Comment (SwatiPeriwal) [inline docs/image-transfer-readme.md:9]: Update the configuration file `config.yml` with Docker Hub credentials and the destination organization.

File: docs/maven-build-readme.md [hunk 6]
```diff
@@ -0,0 +1,66 @@
+# Maven Build Workflow
+
+This workflow is designed to automate the Maven build process for a specific service in the Mosip/Kattu project. It includes tasks related to setting up the build environment, building the project, and uploading artifacts. The workflow can be triggered when certain conditions are met.
```
Comment (SwatiPeriwal) [inline docs/maven-build-readme.md:3]: `Mosip/Kattu`  project

File: docs/maven-build-readme.md [hunk 7]
```diff
@@ -0,0 +1,66 @@
...
+
+## Purpose
+
+The purpose of this workflow is to perform the following tasks:
+- Build the specified service using Maven.
+- Perform checks related to the project's licenses, developers, and plugins.
+- Validate the `pom.xml` and `settings.xml` files.
+- Create an executable JAR and upload it as an artifact.
+- Notify via Slack in case of job failure.
```
Comment (SwatiPeriwal) [inline docs/maven-build-readme.md:12]: Notify via Slack in case of job failures.

File: docs/maven-sonar-analysis-readme.md [hunk 8]
```diff
@@ -0,0 +1,68 @@
...
+- Set up the build environment for a Maven project.
+- Perform SonarCloud analysis on the Maven project.
+- Notify via Slack in case of job failure.
+
+## Inputs
+
+This workflow accepts the following inputs:
+- `SERVICE_LOCATION` (required): The location of the service to be analyzed.
+- `SONAR_URL` (optional, default: 'https://sonarcloud.io'): URL of the Sonar server for analysis.
+- `PROJECT_KEY` (optional, default: "mosip_${{ github.event.repository.name }}"): Project key for Sonar analysis.
```
Comment (SwatiPeriwal) [inline docs/maven-sonar-analysis-readme.md:18]: Leave a space and colon not required after **Secrets**

File: docs/npm-sonar-analysis-readme.md [hunk 9]
```diff
@@ -0,0 +1,62 @@
...
+
+This workflow automates the process of performing SonarQube code analysis on a Node.js project using npm.
+It is designed to help maintain the code quality and identify issues in the codebase.
+The workflow includes steps to install Node.js dependencies, build the project, run Sonar analysis, and send Slack notifications based on the analysis results.
+
+## Inputs
+
+This workflow requires the following inputs:
+
+- `SERVICE_LOCATION` (required): The location of the Node.js project for which you want to run SonarQube analysis.
```
Comment (SwatiPeriwal) [inline docs/npm-sonar-analysis-readme.md:13]: `Node.js`

File: docs/tag-readme.md [hunk 10]
```diff
@@ -0,0 +1,64 @@
+# Tagging of Repos Workflow
+
+## Purpose
+
+This workflow automates the process of creating GitHub releases by applying tags to your repositories.
```
Comment (SwatiPeriwal) [inline docs/tag-readme.md:5]: This workflow automates the process of creating GitHub releases by applying **tags** to your repositories.

**Review by SwatiPeriwal — APPROVED** [review]
Some minor updates... incorporate as needed...

**Review by ckm007 — APPROVED** [review]

**Pull request merged**